File: classlog-backend/src/main/java/com/example/classlog/dto/GradeDto.java
package com.example.classlog.dto;

import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class GradeDto {

  private Long gradeId;
  private ClassDto assignedClass;
  private UserDto student;
  private UserDto teacher;
  private Integer grade;
  private Integer wage;
  private String description;
  private LocalDateTime createdAt;
}

File: classlog-frontend/src/app/model/entities/grade-dto.ts
import {ClassDto} from "./class-dto";
import {UserDto} from "./user-dto";

export interface GradeDto {
  gradeId: number;
  assignedClass: ClassDto;
  student: UserDto;
  teacher: UserDto;
  grade: number;
  wage: number;
  description: string;
  createdAt: Date;
}
